scripts/orig_scripts/knn2img.py

```diff
@@ -390,8 +390,8 @@ def __call__(self, x, n):
                     grid = make_grid(grid, nrow=n_rows)
 
                     # to image
-                    grid = 255.0 * rearrange(grid, "c h w -> h w c").cpu().numpy()
-                    Image.fromarray(grid.astype(np.uint8)).save(os.path.join(outpath, f"grid-{grid_count:04}.png"))
+                    grid_np = 255.0 * rearrange(grid, "c h w -> h w c").cpu().numpy()
+                    Image.fromarray(grid_np.astype(np.uint8)).save(os.path.join(outpath, f"grid-{grid_count:04}.png"))
```

Comment:
Does `flake8` catch reassignment like this? Strongly prefer non-mutating logic if possible; this makes me feel warm and fuzzy inside!

Reply:
Flake8 doesn't but mypy will. However, flake8 did catch this one for some reason that I don't exactly remember. I think it was complaining that a `Tensor` doesn't have an `.astype()` method maybe.